solr/core/src/java/org/apache/solr/handler/loader/XMLLoader.java

```diff
@@ -404,6 +404,7 @@ public SolrInputDocument readDoc(XMLStreamReader parser) throws XMLStreamExcepti
     StringBuilder text = new StringBuilder();
     String name = null;
     boolean isNull = false;
+    boolean isLabeledChildDoc = false;
```

Comment:
I think this won't handle deeply nested cases.  Please enhance the test to have another layer of docs and lets see what happens.

Reply:
I just added a test which seems to pass.
Hopefully I did not miss anything when I wrote it. :)